File: lonely-integer/LonelyInteger.java
public class LonelyInteger {

    /**
     * Detects lonely integer.
     * Example:
     * 1, 1, 2, 3, 3
     * Should return 2.
     * @param integers An integers to verify
     * @return The loneluy integer
     */
    public static int detectLonelyInteger(int[] integers) {
        int value = 0;
        for (int integer : integers) {
            value ^= integer;
        }
        return value;
    }
}

File: operations_over_bits/LonelyInteger.kt
/**
 * Detects lonely integer.
 * Example:
 * Given { 1, 1, 2, 3, 3 }
 * Returns 2
 */
fun findLonelyInteger(integers: IntArray): Int {
    var lonely = 0
    for (integer in integers) {
        lonely = lonely xor integer // java: lonely ^= integer
    }
    return lonely
}
